### PR TITLE
test: Add `claimFunds` test for transfer failure (SC-4387)

### PR DESCRIPTION
**Pull request by vbidin**
# Description

Added missing `claimFunds` test for transfer failure.

# Integrations Checklist

- [ ] Have any function signatures changed? If yes, outline below.
- [ ] Have any features changed or been added? If yes, outline below.
- [ ] Have any events changed or been added? If yes, outline below.
- [ ] Has all documentation been updated?

# Changelog

### CODE REVIEW

File: contracts/test/MapleLoanInternals.t.sol [hunk 1]
```diff
@@ -1065,7 +1065,15 @@ contract MapleLoanInternals_ClaimFundsTests is TestUtils {
         _loan.claimFunds(amountToClaim_, address(this));
     }
 
-    // TODO: testFail_claimFunds_transferFail?
+    function test_claimFunds_transferFail() external {
+        _loan.setClaimableFunds(1);
+
+        try _loan.claimFunds(1, address(this)) {
```
Comment (edag94) [inline contracts/test/MapleLoanInternals.t.sol:1071]: I would leave a short comment here about why this reverts, as its a special case.

**Comment by lucas-manuel** [comment]
Also need to update to title format

File: contracts/test/MapleLoanInternals.t.sol (hunk 1 above)
Comment (lucas-manuel) [reply contracts/test/MapleLoanInternals.t.sol:1071]: Agreed

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: contracts/test/MapleLoanInternals.t.sol (hunk 1 above)
Comment (vbidin) [reply contracts/test/MapleLoanInternals.t.sol:1072]: Affirmative

**Review by lucas-manuel — APPROVED** [review]

**Review by edag94 — APPROVED** [review]

**Pull request merged**